### PR TITLE
Add some notes on how to run tests against a real Artifactory

### PR DESCRIPTION
**Pull request by czoido**
Changelog: Feature: Instructions on how to run conan tests against a real Artifactory server.
Docs: omit

- [ ] Refer to the issue that supports this Pull Request.
- [X] If the issue has missing info, explain the purpose/use case/pain/need that covers this Pull Request.
- [X] I've read the [Contributing guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [X] I've followed the PEP8 style guides for Python code.
- [ ] I've opened another PR in the Conan docs repo to the ``develop`` branch, documenting this one. 

<sup>**Note:** By default this PR will skip the slower tests and will use a limited set of python versions. Check [here](https://github.com/conan-io/conan/blob/develop/.github/PR_INCREASE_TESTING.md) how to increase the testing level by writing some tags in the current PR body text.</sup>

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -208,6 +208,21 @@ To run specific tests, you can specify the test name too, something like:
...
 
+Also, you can run tests against an instance of Artifactory. Those tests should add the attribute
+``artifactory_ready``. Some environment variables have to be defined to run them. For example, for an
+Artifactory instance that is running on the localhost with default user and password configured, the
+variables could take the values:
+
+.. code-block:: bash
+
+    $ export CONAN_TEST_WITH_ARTIFACTORY=1
+    $ export ARTIFACTORY_DEFAULT_URL=http://localhost:8081/artifactory
```
Comment (memsharded) [inline README.rst:225]: It should be clear what URL is it. I guess it is one from a repo, from the SetMeUp? It doesn't look like it

File: README.rst [hunk 2]
```diff
@@ -208,6 +208,21 @@ To run specific tests, you can specify the test name too, something like:
 
 The ``--nocapture`` argument can be useful to see some output that otherwise is captured by nosetests.
 
+Also, you can run tests against an instance of Artifactory. Those tests should add the attribute
```
Comment (memsharded) [inline README.rst:211]: Maybe it would be good to actually add a command line example like ``nosetests . -A "artifactory_ready"``?

**Review by memsharded — APPROVED** [review]

File: README.rst (hunk 1 above)
Comment (czoido) [reply README.rst:225]: This is not the url for the repo but the base url, the tests will create repos inside this artifactory instance to run. Maybe I can add a note about that

Comment (memsharded) [reply README.rst:225]: Understood. As you see, not really necessary, it is correct, it was just me with some false assumptions.

**Review by memsharded — APPROVED** [review]

**Pull request merged**